### ISSUE TITLE
Network Variables preventing the game from launching

### ISSUE THREAD

**Issue by zeeblo** [Xilophor/LethalNetworkAPI#30]
I started a brand new test file to properly check this and according to the documentation (granted outdated currently but I assume the concepts should be the same) network variables should be able to contain any serializable type (string, int, etc.)

The problem I'm noticing is that on version 3.3.1 the game doesn't allow bools or even floats

This is the whole code.

```cs
using BepInEx;
using BepInEx.Logging;
using HarmonyLib;
using LethalNetworkAPI;
using static BepInEx.BepInDependency;

namespace MyPJ
{

    [BepInPlugin(modGUID, modName, modVersion)]
    [BepInDependency(LethalNetworkAPI.MyPluginInfo.PLUGIN_GUID, DependencyFlags.HardDependency)]
    public class Plugin : BaseUnityPlugin
    {
        private const string modGUID = "zeeblo.MyPJ";
        private const string modName = "MyPJ";
        private const string modVersion = "0.1.0";
        private readonly Harmony _harmony = new(modGUID);
        public static Plugin? instance;
        internal static ManualLogSource mls = BepInEx.Logging.Logger.CreateLogSource(modGUID);
        public static LNetworkVariable<bool> v1 = LNetworkVariable<bool>.Connect("v1");

        private void Awake()
        {
            Plugin.mls.LogInfo(">>> Started Mod");
        }


    }
}
```

OUTPUT
```
[Error  : Unity Log] NullReferenceException: Object reference not set to an instance of an object
Stack trace:
LethalNetworkAPI.LNetworkVariable`1[TData].ReceiveUpdate (System.Object data) (at ./Public/LNetworkVariable.cs:228)
LethalNetworkAPI.Internal.UnnamedMessageHandler.HandleMessage (System.UInt64 clientId, System.String messageID, LethalNetworkAPI.Internal.EMessageType messageType, System.Object messageData, System.UInt64[] targetClients) (at ./Internal/UnnamedMessageHandler.cs:313)
LethalNetworkAPI.Internal.UnnamedMessageHandler.SendMessageToClients (LethalNetworkAPI.Internal.MessageData messageData, System.UInt64[] clientGuidArray, System.Boolean deprecatedMessage) (at ./Internal/UnnamedMessageHandler.cs:140)
LethalNetworkAPI.Internal.UnnamedMessageHandler.UpdateNewClientVariables (System.UInt64 newClient) (at ./Internal/UnnamedMessageHandler.cs:101)
Unity.Netcode.NetworkConnectionManager.InvokeOnClientConnectedCallback (System.UInt64 clientId) (at <895801699cfc4b4ab52267f31e2a4998>:0)
Unity.Netcode.NetworkManager.HostServerInitialize () (at <895801699cfc4b4ab52267f31e2a4998>:0)
Unity.Netcode.NetworkManager.StartHost () (at <895801699cfc4b4ab52267f31e2a4998>:0)
UnityEngine.Debug:LogException(Exception)
Unity.Netcode.NetworkManager:StartHost()
MenuManager:StartHosting()
<StartHost>d__87:MoveNext()
System.Runtime.CompilerServices.AsyncVoidMethodBuilder:Start(<StartHost>d__87&)
GameNetworkManager:StartHost()
MenuManager:ConfirmHostButton()
UnityEngine.EventSystems.EventSystem:Update()
```

It completely stops me from joining my own lobby and throws that error in the console.

Now I could obviously just use a string and convert it to a float/bool where necessary since I'm noticing that a string seems to be the only reliable TData but I thought I'd bring this to your attention just incase you weren't aware of this issue

**Comment by Fandovec03** [comment]
I get the same error message whenever host disconnects and tries to rehost. I use variables to have synced enemy values between clients and it works without any issues.

**Comment by Xilophor** [comment]
Resolved in v3.3.2
